da/celestia/mock/server.go

```diff
@@ -70,137 +101,134 @@ func (s *Server) Stop() {
+	var req request
 	err := json.NewDecoder(r.Body).Decode(&req)
 	if err != nil {
 		s.writeError(w, err)
 		return
 	}
-
-	block := types.Block{}
-	blockData, err := hex.DecodeString(req.Data)
-	if err != nil {
-		s.writeError(w, err)
-		return
-	}
-	err = block.UnmarshalBinary(blockData)
-	if err != nil {
-		s.writeError(w, err)
-		return
-	}
-
-	res := s.mock.SubmitBlock(r.Context(), &block)
-	code := 0
-	if res.Code != da.StatusSuccess {
-		code = 3
-	}
-
-	resp, err := json.Marshal(cnc.TxResponse{
-		Height: int64(res.DAHeight),
-		Code:   uint32(code),
-		RawLog: res.Message,
-	})
-	if err != nil {
-		s.writeError(w, err)
-		return
-	}
-
-	s.writeResponse(w, resp)
-}
-
-func (s *Server) shares(w http.ResponseWriter, r *http.Request) {
-	height, err := parseHeight(r)
-	if err != nil {
-		s.writeError(w, err)
-		return
-	}
-
-	res := s.mock.RetrieveBlocks(r.Context(), height)
-	if res.Code != da.StatusSuccess {
-		s.writeError(w, errors.New(res.Message))
-		return
-	}
-
-	var nShares []NamespacedShare
-	for _, block := range res.Blocks {
-		blob, err := block.MarshalBinary()
+	switch req.Method {
+	case "header.GetByHeight":
+		var params []interface{}
+		err := json.Unmarshal(req.Params, &params)
 		if err != nil {
 			s.writeError(w, err)
 			return
 		}
-		delimited, err := marshalDelimited(blob)
+		height := uint64(params[0].(float64))
+		dah := s.mock.GetHeaderByHeight(height)
+		resp := &response{
+			Jsonrpc: "2.0",
+			Result: header.ExtendedHeader{
+				DAH: dah,
+			},
+			ID:    req.ID,
+			Error: nil,
+		}
+		bytes, err := json.Marshal(resp)
 		if err != nil {
 			s.writeError(w, err)
+			return
 		}
-		nShares = appendToShares(nShares, []byte{1, 2, 3, 4, 5, 6, 7, 8}, delimited)
-	}
-	shares := make([]Share, len(nShares))
-	for i := range nShares {
-		shares[i] = nShares[i].Share
-	}
-
-	resp, err := json.Marshal(namespacedSharesResponse{
-		Shares: shares,
-		Height: res.DAHeight,
-	})
-	if err != nil {
-		s.writeError(w, err)
-		return
-	}
-
-	s.writeResponse(w, resp)
-}
-
-func (s *Server) data(w http.ResponseWriter, r *http.Request) {
-	height, err := parseHeight(r)
-	if err != nil {
-		s.writeError(w, err)
-		return
-	}
-
-	res := s.mock.RetrieveBlocks(r.Context(), height)
-	if res.Code != da.StatusSuccess {
-		s.writeError(w, errors.New(res.Message))
-		return
-	}
-
-	data := make([][]byte, len(res.Blocks))
-	for i := range res.Blocks {
-		data[i], err = res.Blocks[i].MarshalBinary()
+		s.writeResponse(w, bytes)
+	case "blob.GetAll":
+		var params []interface{}
+		err := json.Unmarshal(req.Params, &params)
+		if err != nil {
+			s.writeError(w, err)
+			return
+		}
+		height := params[0].(float64)
+		block := s.mock.RetrieveBlocks(r.Context(), uint64(height))
+		var blobs []blob.Blob
+		for _, block := range block.Blocks {
+			data, err := block.MarshalBinary()
+			if err != nil {
+				s.writeError(w, err)
+				return
+			}
+			ns, err := share.NewBlobNamespaceV0([]byte{0x00, 0x01, 0x02, 0x03, 0x04})
```

Comment:
why `0x00, 0x01, 0x02, 0x03, 0x04` for namespace? does `params` not pass the namespace array? any issue using that?

Reply:
Fixed.